src/scaler.rs

```diff
@@ -45,24 +45,22 @@ pub fn stretch(
 }
 
 pub fn zoom(img: &image::DynamicImage, layer_width: u32, layer_height: u32) -> image::DynamicImage {
-    let (w, h) = (img.width(), img.height());
-
-    let ratio = (layer_width as f64 / w as f64).max(layer_height as f64 / h as f64);
-
-    let (new_width, new_height) = (
-        (w as f64 * ratio).round() as u32,
-        (h as f64 * ratio).round() as u32,
+    // TODO HDR?
+    let img = img.to_rgba8();
+    let pixmap = tiny_skia::PixmapRef::from_bytes(&img, img.width(), img.height()).unwrap();
+    let mut target_pixmap = tiny_skia::Pixmap::new(layer_width, layer_height).unwrap();
+    target_pixmap.draw_pixmap(
+        0,
+        0,
+        pixmap,
+        &tiny_skia::PixmapPaint::default(),
+        tiny_skia::Transform::from_scale(
+            layer_width as f32 / img.width() as f32,
+            layer_height as f32 / img.height() as f32,
+        ),
+        None,
     );
-
-    let mut new_image = image::imageops::resize(img, new_width, new_height, FilterType::Lanczos3);
```

Comment:
The Lanczos3 filter was choosen in particular because of it's great quality scaling images without matching resolution. tiny_skia's filter is just doing linear in comparison and will look much worse.

Reply:
See also: https://docs.rs/image/0.25.2/image/imageops/enum.FilterType.html

Reply:
Ah right. I wonder if a different parameters to Lanczos / a different implementation / another higher quality scaling method could perform a little better. The amount of time it currently takes is a bit excessive for a semi-realtime use case.

Caching the image from the file and avoiding drawing more times than needed should at least help, either way.

Reply:
Apparently (https://github.com/image-rs/image/issues/2223) the `image` crate is known to have somewhat poor performance for this, and the `fast_image_resize` crate has a much faster SIMD implementation, including for Lanczos3. So we should be able to improve this relatively easily without sacrificing quality.